src/pages/ReviewPage/ReviewPage.js

```diff
@@ -37,13 +43,39 @@ export default class ReviewPage extends Component {
         }
       })
       .then(res => {
-        const updatedMovieData = res['result'].slice(
-          movieData.length,
-          movieData.length + 7
-        );
-        this.setState({
-          movieData: [...movieData, ...updatedMovieData],
-        });
+        let keyName = id ? 'genre_movie' : 'movie_random';
+        if (!id) {
+          const updatedMovieData = res[keyName].slice(
+            randomMovie.length,
+            randomMovie.length + 7
+          );
+          this.setState(
+            {
+              randomMovie: [...randomMovie, ...updatedMovieData],
+            },
+            () => {
+              this.setState({
+                movieData: [...randomMovie],
+              });
+            }
```

Comment:
setState 두번 해줄 필요 없어보입니다.
```suggestion
            {
              randomMovie: [...randomMovie, ...updatedMovieData],
              movieData:[...randomMovie, ...updatedMovieData],
            }
```
로 해줘도 될 것 같네요